ushahidi/platform-client-mzima
# [Outreachy Task Submission] Add Hidden Button Value Text for Login & Signup Screens

# Introduction
The "make password visible" button icon in both the `Login` and `Signup` view has a problem. The button has only an icon and has no value text. While this is just a stylistic choice, it raises an issue with any accessibility checking tool.  It does this because a button having no value text is a violation of WCAG Success Criterion 1.1.1 (Level A) - Non-text content. [See](https://www.w3.org/WAI/WCAG21/Understanding/non-text-content.html). See a screenshot of the pertinent component:

![image](https://github.com/ushahidi/platform-client-mzima/assets/29470516/86b9e2e4-fa78-4919-b22e-bee64d6a75ba)
> ***Note the SVG icon***

# How to Test the PR
## Manually Testing
1. Open your browser and log into your running Ushahidi deployment (from localhost).
2. Navigate to either the `login & signup` section.
3. Try right-clicking on the icon at the right side of the password input field of the screen and selecting 'inspect'.
4. View the properties. A few lines above the currently highlighted snippet, look for a `span` element with a class of `mzima-button__text`. See screenshot below:
![image](https://github.com/ushahidi/platform-client-mzima/assets/29470516/31a4bd15-d8c3-40b7-9fca-108a0b91fc56)
5. You will not see any text in it.
6. Stop the running Ushahidi deployment in your terminal.
7. Checkout to this PR branch.
8. Re-run your Ushahidi deployment.
10. Repeat steps 1-4.
11. You will see a value in the `span` element this time around, and this text will be visually hidden.

## Using an Accessibility Checker
If you use a tool like [WAVE evaluation tool](https://chromewebstore.google.com/detail/wave-evaluation-tool/jbbplnpkjmmeebjpijfedlgcdilocofh), it's as simple as following steps 1 and 2 above, and then turning on the browser extension. It will flag the issue. Checkout to the PR and do this again, you will notice the issue is no longer flagged. 

## Dependency
This fix has a dependency on a previous [PR](https://github.com/ushahidi/platform-client-mzima/pull/860)

## Fixes
This fixes [#4757](https://github.com/ushahidi/platform/issues/4757)
